consensus/polybft/exit_relayer.go

```diff
@@ -0,0 +1,300 @@
+package polybft
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi"
+	"github.com/0xPolygon/polygon-edge/txrelayer"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/umbracle/ethgo"
+)
+
+var errUnknownExitEvent = errors.New("unknown event from exit helper or checkpoint manager")
+
+// ExitRelayer is an interface that defines functions of an exit event handler and executioner
+type ExitRelayer interface {
+	Close()
+	Init() error
+	AddLog(eventLog *ethgo.Log) error
+	PostBlock(req *PostBlockRequest) error
+}
+
+var _ ExitRelayer = (*dummyExitRelayer)(nil)
+
+type dummyExitRelayer struct{}
+
+func (d *dummyExitRelayer) Close()                                {}
+func (d *dummyExitRelayer) Init() error                           { return nil }
+func (d *dummyExitRelayer) AddLog(eventLog *ethgo.Log) error      { return nil }
+func (d *dummyExitRelayer) PostBlock(req *PostBlockRequest) error { return nil }
+
+// EventProofRetriever is an interface that exposes function for retrieving exit proof
+type EventProofRetriever interface {
+	GenerateExitProof(exitID uint64) (types.Proof, error)
+}
+
+var _ ExitRelayer = (*exitRelayer)(nil)
+
+// exitRelayer handles checkpoint submitted events and executes exit events
+type exitRelayer struct {
+	relayerEventsProcessor
+
+	key            ethgo.Key
+	proofRetriever EventProofRetriever
+	txRelayer      txrelayer.TxRelayer
+	logger         hclog.Logger
+	exitStore      *ExitStore
+
+	notifyCh chan struct{}
+	closeCh  chan struct{}
+
+	exitHelperAddr types.Address
+}
+
+// newExitRelayer creates a new instance of exitRelayer
+func newExitRelayer(
+	txRelayer txrelayer.TxRelayer,
+	key ethgo.Key,
+	proofRetriever EventProofRetriever,
+	blockchain blockchainBackend,
+	exitStore *ExitStore,
+	config *relayerConfig,
+	exitHelperAddr types.Address,
+	logger hclog.Logger) *exitRelayer {
+	if config == nil {
+		config = &relayerConfig{
+			maxBlocksToWaitForResend: defaultMaxBlocksToWaitForResend,
+			maxAttemptsToSend:        defaultMaxAttemptsToSend,
+			maxEventsPerBatch:        defaultMaxEventsPerBatch,
+		}
+	}
+
+	relayer := &exitRelayer{
+		key:            key,
+		logger:         logger,
+		exitStore:      exitStore,
+		txRelayer:      txRelayer,
+		proofRetriever: proofRetriever,
+		exitHelperAddr: exitHelperAddr,
+		closeCh:        make(chan struct{}),
+		notifyCh:       make(chan struct{}, 1),
+	}
+
+	relayerEventsProcessor := relayerEventsProcessor{
+		config:     config,
+		logger:     logger,
+		state:      exitStore,
+		blockchain: blockchain,
+		sendTx:     relayer.sendTx,
+	}
+
+	relayer.relayerEventsProcessor = relayerEventsProcessor
```

Comment:
Inline it and in case `relayerEventsProcessor` gets changed to a pointer type, then instantiate it with an &:

```suggestion
	relayer.relayerEventsProcessor = &relayerEventsProcessor{
		config:     config,
		logger:     logger,
		state:      exitStore,
		blockchain: blockchain,
		sendTx:     relayer.sendTx,
	}
```

Reply:
Fix: https://github.com/Ethernal-Tech/blade/pull/47/commits/d439dc0c3031276a74d678e4b2b915ca9fd20f74